### PR TITLE
Open up FileSpec.Builder for more mutation

### PR DESCRIPTION
**Pull request by ZacSweers**
Similar towhat we do in other spec builders, this allows mutation of builder bits directly (so it's no longer just additive).

This covers:
- `comment`
- `members`
- `memberImports`. Specifically - this makes `Import` public but keeps its constructor `internal` and adds an adder. This allows reading Imports and re-adding them later if they're cleared in between, and seemed strictly better than just making the `memberImports` property itself public since we want folks to use the `addImport` functions.

### CODE REVIEW

File: kotlinpoet/src/test/java/com/squareup/kotlinpoet/FileSpecTest.kt [hunk 1]
```diff
@@ -754,6 +754,52 @@ class FileSpecTest {
...
+
+  @Test fun modifyMembers() {
+    val builder = FileSpec.builder("com.taco", "Taco")
+        .addFunction(FunSpec.builder("aFunction").build())
+        .addFunction(FunSpec.builder("aSecondFunction").build())
+        .addType(TypeSpec.classBuilder("AClass").build())
+
+    builder.members.removeAll { it !is TypeSpec }
+
+    check(builder.build().members.all { it is TypeSpec })
```
Comment (Egorand) [inline kotlinpoet/src/test/java/com/squareup/kotlinpoet/FileSpecTest.kt:789]: Maybe `assertThat(builder.build().members).containsExactly(TypeSpec.classBuilder("AClass").build())`?

File: kotlinpoet/src/test/java/com/squareup/kotlinpoet/FileSpecTest.kt [hunk 2]
```diff
@@ -754,6 +754,52 @@ class FileSpecTest {
...
+  }
+
+  @Test fun clearIndent() {
```
Comment (Egorand) [inline kotlinpoet/src/test/java/com/squareup/kotlinpoet/FileSpecTest.kt:792]: nit: `clearComment`

File: kotlinpoet/src/test/java/com/squareup/kotlinpoet/FileSpecTest.kt (hunk 1 above)
Comment (ZacSweers) [reply kotlinpoet/src/test/java/com/squareup/kotlinpoet/FileSpecTest.kt:789]: oops, I meant to actually add a typealias and property too to this test

File: kotlinpoet/src/test/java/com/squareup/kotlinpoet/FileSpecTest.kt (hunk 2 above)
Comment (ZacSweers) [reply kotlinpoet/src/test/java/com/squareup/kotlinpoet/FileSpecTest.kt]: 755ccae

File: kotlinpoet/src/test/java/com/squareup/kotlinpoet/FileSpecTest.kt (hunk 1 above)
Comment (ZacSweers) [reply kotlinpoet/src/test/java/com/squareup/kotlinpoet/FileSpecTest.kt:790]: 755ccae

**Pull request merged**